error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/AssertJThrowingCallableRules.java

```diff
@@ -759,4 +759,19 @@ static final class AbstractThrowableAssertWithFailMessage {
       return abstractThrowableAssert.withFailMessage(message, args);
     }
   }
+
+  static final class AbstractThrowableAssertCauseIsSameAs {
+    @BeforeTemplate
+    @SuppressWarnings("deprecation" /* Migrates away from deprecated API. */)
+    AbstractThrowableAssert<?, ? extends Throwable> before(
+        AbstractThrowableAssert<?, ? extends Throwable> throwableAssert, Throwable expected) {
+      return throwableAssert.hasCauseReference(expected);
+    }
+
+    @AfterTemplate
+    AbstractThrowableAssert<?, ? extends Throwable> after(
+        AbstractThrowableAssert<?, ? extends Throwable> throwableAssert, Throwable expected) {
+      return throwableAssert.cause().isSameAs(expected);
+    }
+  }
```

Comment:
Other rules in this class suffer the same issue, so not a deal-breaker, but note that this rewrite changes semantics: subsequent chained assertions will now be applied to the cause, rather than the top-level exception. All add an `XXX` comment calling this out.